worlds/checksmate/test/TestPieceLimits.py

```diff
@@ -0,0 +1,173 @@
+from . import CMTestBase
+
+
+# I don't like that this generates many entire seeds just to check some global logic.
+# TODO(chesslogic): find_piece_limit should accept an army option value. Store piece distribution on some other helper
+
+
+class PieceLimitTestBase(CMTestBase):
+
+    def world_setup(self, *args, **kwargs):
+        super().world_setup(*args, **kwargs)
+        self.NO_CHILDREN = self.world.PieceLimitCascade.NO_CHILDREN
+        self.ACTUAL_CHILDREN = self.world.PieceLimitCascade.ACTUAL_CHILDREN
+        self.POTENTIAL_CHILDREN = self.world.PieceLimitCascade.POTENTIAL_CHILDREN
```

Comment:
I'd recommend a different style:

in the class body do `NO_CHICKEN = CMWorld.PieceLimitCascade.NO_CHILDREN`, etc.

The way this is written is missing type annotation and also violates variable naming since those are technically non-constants because of the way you define them. After `pip install mypy`, you can run `python -m mypy worlds/checksmate` or even `python -m mypy --strict worlds/checksmate` to see type errors.